File: course/CSCI451/2015Fall/src/vigeneretableau/VigenereTable.java
public class VigenereTable {
    private final int ALPHABET_SIZE = 26;
    private final int KEY_SPACE_SIZE = 26;

    public static void main(String[] args) {
        VigenereTable vt = new VigenereTable();
        vt.printTable();
    }

    private void printTable() {
        int alphabet[] = new int[ALPHABET_SIZE];
        int keyspace[] = new int[KEY_SPACE_SIZE];

        // initialize alphabet
        for (int i = 0; i < ALPHABET_SIZE; i ++) {
            alphabet[i] = i;
        }

        // initialize key space
        for (int j = 0; j < KEY_SPACE_SIZE; j ++) {
            keyspace[j] = j;
        }

        // output table header
        System.out.print("  ");
        for (int i = 0; i < ALPHABET_SIZE; i ++) {
            System.out.print((char)(alphabet[i] + 'A') + " "); 
        }
        System.out.println();

        // compute and output the row for a key letter
        for (int key = 0; key < KEY_SPACE_SIZE; key ++) {
            System.out.print((char)(key + 'A') + " "); 
            for (int letter = 0; letter < ALPHABET_SIZE; letter ++) {
                int ciphertext = (letter + key) % ALPHABET_SIZE;
                System.out.print((char)(ciphertext + 'A') + " ");
            }
            System.out.println();
        }
    }
}

File: course/CSCI451/2015Fall/src/vigeneretableau/vigeneretable.c
#include <stdio.h>

#define ALPHABET_SIZE   26
#define KEY_SPACE_SIZE  26
int main()
{
    int alphabet[ALPHABET_SIZE];
    int keyspace[KEY_SPACE_SIZE];
    int i, j, key, letter, ciphertext;

    // initialize alphabet
    for (i = 0; i < ALPHABET_SIZE; i ++) {
        alphabet[i] = i;
    }

    // initialize key space
    for (j = 0; j < KEY_SPACE_SIZE; j ++) {
        keyspace[j] = j;
    }


    // output table header
    printf("  ");
    for (i = 0; i < ALPHABET_SIZE; i ++) {
        printf("%c ", alphabet[i] + 'A');
    }
    printf("\n");

    // compute and output the row for a key letter
    for (key = 0; key < KEY_SPACE_SIZE; key ++) {
        printf("%c ", key + 'A'); 
        for (letter = 0; letter < ALPHABET_SIZE; letter ++) {
            ciphertext = (letter + key) % ALPHABET_SIZE;
            printf("%c ", ciphertext + 'A');
        }
        printf("\n");
    }

    return 0;
}
